### PR TITLE
chore: moved error management to exchangeSDK

### PR DESCRIPTION
**Pull request by CremaFR**
Moved errors management from swap live app to exchangeSDK in order to have consistent experience between moonpay and demo0

### CODE REVIEW

**Review by sprohaszka-ledger — COMMENTED** [review]
This would be great to start splitting this sdk file in different "sub file" for each different responsibility.

This error management can be defined in another file (or reuse the existing one `error` - but not totally convince).

File: lib/src/sdk.ts [hunk 1]
```diff
@@ -1,14 +1,12 @@
 import {
   Account,
-  CardanoTransaction,
-  CryptoCurrency,
-  Currency,
   ElrondTransaction,
-  NearTransaction,
-  NeoTransaction,
   RippleTransaction,
```
Comment (sarneijim) [inline lib/src/sdk.ts:4]: This library is common for any kind of exchange, not sure why we are removing some families.

File: lib/src/sdk.ts [hunk 2]
```diff
@@ -156,6 +186,35 @@ export class ExchangeSDK {
...
+          case "SwapCompleteExchangeError":
+            break;
+          default:
+            this.walletAPI.custom.swap.throwExchangeErrorToLedgerLive(error);
+            break;
+        }
+      } else if (error instanceof Error) {
+        this.walletAPI.custom.swap.throwGenericErrorToLedgerLive(error);
+      }
+      throw error; // Sentry track
```
Comment (sarneijim) [inline lib/src/sdk.ts:213]: You can remove this comment, as the user can not have implemented Sentry

File: lib/src/sdk.ts [hunk 3]
```diff
@@ -66,9 +65,39 @@ const ExchangeType = {
...
+  }
+}
+
+/**
+ * Should match our swap live apps path naming  :/
+ */
 function getCustomModule(client: WalletAPIClient) {
   return {
     exchange: new ExchangeModule(client),
+    swap: new CustomErrorModule(client),
```
Comment (sarneijim) [inline lib/src/sdk.ts:99]: Do we must move this Module to a different file as it is swap specific? (now, I am thinking...I am not sure if we are in Moonpay context that we will have access to this drawer)

File: lib/src/sdk.ts (hunk 1 above)
Comment (CremaFR) [reply lib/src/sdk.ts:4]: Not used in the file, I auto removed not used imports

**Review by sprohaszka-ledger — COMMENTED** [review]
Only one real comment, apart the ones with **nitpicking** prefix ;)

**Review by sprohaszka-ledger — APPROVED** [review]

**Pull request merged**